lib/pages/saved_places/saved_places.dart

```diff
@@ -0,0 +1,155 @@
+                                // TODO translate
+                                child: Text('Custom Places', style: theme.textTheme.bodyText1),
+                              ),
+                            Column(
+                              children: searchLocationsCubit.state.myPlaces
+                                  .map(
+                                    (place) => LocationTiler(
+                                        location: place,
+                                        enableSetIcon: true,
+                                        enableSetName: true,
+                                        enableSetPosition: true,
+                                        updateLocation: searchLocationsCubit.updateMyPlace,
+                                        removeLocation: searchLocationsCubit.deleteMyPlace),
+                                  )
+                                  .toList(),
+                            )
+                          ],
+                        );
+                      },
+                    ),
+                    floatingActionButton: FloatingActionButton(
+                      onPressed: () {
+                        _addNewPlace(context);
+                      },
+                      backgroundColor: Theme.of(context).primaryColor,
+                      heroTag: null,
+                      child: const Icon(Icons.add),
+                    ),
+                  ),
+                  BlocBuilder<SearchLocationsCubit, SearchLocationsState>(
+                    builder: (context, state) {
+                      return ListView.builder(
+                        padding: const EdgeInsets.all(16.0),
+                        itemCount: state.favoritePlaces.length,
+                        itemBuilder: (BuildContext context, int index) {
+                          return LocationTiler(
+                            location: state.favoritePlaces[index],
+                            enableSetIcon: true,
+                            updateLocation: searchLocationsCubit.updateFavoritePlace,
+                            removeLocation: searchLocationsCubit.deleteFavoritePlace,
+                          );
+                        },
+                      );
+                    },
+                  ),
+                ],
+              ),
+            ),
+          ],
+        ),
+      ),
+      drawer: const TrufiDrawer(SavedPlacesPage.route),
+    );
+  }
+
+  Future<void> _addNewPlace(BuildContext context) async {
+    final searchLocationsCubit = context.read<SearchLocationsCubit>();
+    final LatLng mapLocation = await ChooseLocationPage.selectPosition(context);
+    if (mapLocation != null) {
+      final String description = await showDialog(
+        context: context,
+        builder: (BuildContext context) {
+          return const DialogEditText();
+        },
+      );
+      if (description != null) {
```

Comment:
This case can never happen because it seems that the save button prevents a null value.

Reply:
When we use the cancel button or close the DialogEditText, the description is null.